posts/2020/10/manylinux1-manylinux2010.md

```diff
@@ -0,0 +1,83 @@
+Unixes, you can package your python c-extension packages with a "universal
+platform tag" called [manylinux](https://github.com/pypa/manylinux). The first
+manylinux standard, manylinux1 was based on CentOS5 which has [been
+obsolete](https://endoflife.software/operating-systems/linux/centos) since
+March, 2017. The subesquent manylinux2010 standard, based on CentOS6, will hit
+end-of-life in December 2020. The manylinux2014 standard still has some
+breathing room, based on CentOS7 it will reach end-of-life in July 2024.
+
+## If it is obsolete, why are there still manylinux1 wheels?
+
+Wheels are typically consumed by `pip` via `pip install`. Manylinux wheels are
+typically used for projects that require compilation, otherwise they would ship
+pure python wheels with the "none" platform tag, meaning they are compatible with
+any platform. So say you are a library author and want to make it convenient
+for users to install your package. If you ship a manylinux2014 wheel, but the
+version of pip your users have is too old to support manylinux2014 wheels, pip
+will happily download the source package and compile it for them. Havoc ensues:
+windows users typically cannot compile, prerequisites will be missing.
+
+Pip began supporting manylinux2010 wheels with [version
+19.0](https://github.com/pypa/pip/blob/master/NEWS.rst#190-2019-01-22),
+released in Jan 2019. The version of pip officially shipped with Python3.6, via
+the [ensurepip](https://docs.python.org/3.6/library/ensurepip.html) module, is
+version 18. Python3.7 ships pip 20. It is easy enough to upgrade, but to be on
+the safe side, and prevent havoc, library authors will ship a manylinux1 wheel
+for Python3.6 support.
+
+## What happens now that Python3.6 is falling out of favor?
+[Python 3.6 is no longer in active
+development](https://www.python.org/dev/peps/pep-0494). In fact, the scientific
+python data community has proposed to stop actively supporting Python3.6 [from
+July
+2020](https://numpy.org/neps/nep-0029-deprecation_policy.html#support-table).
+So I would expect to see projects begin to drop the older manylinux1 format,
+and drop support for Python3.6.
+
+## What about anaconda/miniconda and conda-forge?
+Conda does not use the same kind of wheel format provided by PyPI and `pip`. Their
+build system is internally consistent and they build a binary package for each
+OS they support, thus they are not bound to the manylinux designation.
+XXXX ??? Is this accurate? Does conda-forge have a declared Python3.6 phase-out timeline?
+
+## What comes after manylinux2014?
+The glibc used in manylinux2014 is defined as the one used by CentOS7. This OS
+was released in June 2014. This manylinux standard, for the first time,
+declared support for non-x86 hardware systems like ARM64 (aarch64), Power
+(ppc64) and S390X.  However the ARM platform has grown greatly since 2014, and
+glibc has moved from version 2.17 to 2.31, fixing many bugs. Since the real
```

Comment:
Around here may be a useful place to mention that if you are on ARM64 today, using Miniforge may be a good idea, because conda-forge has much better `aarch64` support than PyPI right now.

Reply:
added